File: ContainerWithMostWater.py
class Solution:
    def maxArea(self, height: list[int]) -> int:
        maxarea = 0
        for left in range(len(height)):
            for right in range(left + 1, len(height)):
                width = right - left
                maxarea = max(maxarea, min(height[left], height[right]) * width)

        return maxarea
    


a = Solution()
print(a.maxArea([1, 8, 6, 4, 7, 6, 3, 2]))

File: ContainerWithMostWater.ts
function maxArea(height: number[]): number {
    let maxarea = 0;

    for (let left = 0; left < height.length; left++) {
        for (let right = left + 1; right < height.length; right++) {
            let width = right - left;
            maxarea = Math.max(
                maxarea,
                Math.min(height[left], height[right]) * width,
            );
        }
    }

    return maxarea;
}

console.log(maxArea([1, 8, 6, 4, 7, 6, 3, 2]))